agent/command/generate.go

```diff
@@ -103,14 +103,21 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 		pollRetryMaxDelay = 15 * time.Second
 	)
 
-	var generateStatus *apimodels.GeneratePollResponse
 	err = utility.Retry(
 		ctx,
 		func() (bool, error) {
-			generateStatus, err = comm.GenerateTasksPoll(ctx, td)
+			generateStatus, err := comm.GenerateTasksPoll(ctx, td)
 			if err != nil {
 				return true, err
 			}
+			// only retry if the errors are with saving the project
+			if len(generateStatus.Errors) > 0 {
+				fullErr := errors.New(strings.Join(generateStatus.Errors, ", "))
```

Comment:
I think constructing an error here is redundant, since you grab the string with `.Error()` later. I think instead you can
```
fullErr := strings.Join(generateStatus.Errors, ", ")
if strings.Contains(fullErr, evergreen.RetryGenerateTasksError) {
```

Reply:
that's fair, we do still return fullErr as the error though so I figured calling .Error once was cleaner than calling errors.New(fullError) twice

Reply:
Oh, I missed we return the error later.